src/constants/formComponentTemplate.tsx

```diff
@@ -0,0 +1,155 @@
+import { JSX } from "react";
+
+import { AssetPicker } from "@/components/FormElements/AssetPicker";
+import { PubKeyPicker } from "@/components/FormElements/PubKeyPicker";
+import { OrderPicker } from "@/components/FormElements/OrderPicker";
+import { CursorPicker } from "@/components/FormElements/CursorPicker";
+import { LimitPicker } from "@/components/FormElements/LimitPicker";
+
+import { parseJsonString } from "@/helpers/parseJsonString";
+import { validate } from "@/validate";
+import { AnyObject, AssetObjectValue } from "@/types/types";
+
+type TemplateRenderProps = {
+  value: string | undefined;
+  error: string | undefined;
+  // eslint-disable-next-line no-unused-vars
+  onChange: (val: any) => void;
+  isRequired?: boolean;
+};
+
+type TemplateRenderAssetProps = {
+  value: AssetObjectValue | undefined;
+  error: { code: string | undefined; issuer: string | undefined } | undefined;
+  // eslint-disable-next-line no-unused-vars
+  onChange: (asset: AssetObjectValue | undefined) => void;
+  isRequired?: boolean;
+};
+
+type TemplateRenderOrderProps = {
+  value: string | undefined;
+  // eslint-disable-next-line no-unused-vars
+  onChange: (optionId: string | undefined, optionValue?: string) => void;
+  isRequired?: boolean;
+};
+
+type FormComponentTemplate = {
+  // eslint-disable-next-line no-unused-vars
+  render: (...args: any[]) => JSX.Element;
+  // eslint-disable-next-line no-unused-vars
+  validate: ((...args: any[]) => any) | null;
+};
+
+export const formComponentTemplate = (
```

Comment:
All the components are rendered from here. We'll add more as we go.

Reply:
Sounds good. I'll add mine to `FormElements`

Reply:
is this file supposed to be in `constants`?

Reply:
I wasn't sure where else to put it. :woman_shrugging: It's static data that we render dynamically. I'm open to suggestions. :pray:

Reply:
it feels more like a component to me because it imports `components`.

Reply:
:+1: I'll move it to `components` in the next PR.